src/JukeBox.cpp

```diff
@@ -30,49 +49,284 @@ bool JukeBox::start(string exeName, string configFilePath) {
 
   initLoggingHandler(exeName);
   DLOG(INFO) << "Hello world from JukeBox main.cpp !";
+
+  mNetwork->handleRequests();
+
   return true;
 }
 
-TResult<TSessionID> JukeBox::generateSession(
-    optional<TPassword> const &pw, std::optional<std::string> const &nickname) {
-  return Error(ErrorCode::NotImplemented,
-               "generateSession is not implemented yet");
+TResult<TSessionID> JukeBox::generateSession(optional<TPassword> const &pw,
+                                             optional<string> const &nickname) {
+  static int userID = 0;
+  User user;
+  auto conf = ConfigHandler::getInstance();
+  auto adminPw = conf->getValueString("MainParams", "adminPassword");
+  if (holds_alternative<Error>(adminPw))
+    return get<Error>(adminPw);
+
+  string name = "(no nickname given)";
+  if (nickname.has_value())
+    name = nickname.value();
+  user.Name = name;
+
+  if (pw.has_value() && pw.value() == get<string>(adminPw)) {
+    LOG(INFO) << "JukeBox.generateSession: User '" << name << "' is admin!";
+    user.isAdmin = true;
+  }
+
+  /* Generate a unique ID, consisting of a counter and the number of seconds
+   * since 1970 */
+  user.SessionID = "ID" + to_string(userID) + to_string(time(nullptr));
```

Comment:
the handling of `userID` has to be locked, otherwise we have a race condition again. something like this should do it:
```c++
static mutex userMutex;

{
    unique_lock lck(userMutex);
    user.SessionID = ...
    userID++;
}
```